Publish minutes of 2022-11-10 meeting (#321)
hober/web-extensions: _minutes/2022-11-10-wecg.md

## Attendees (sign yourself in)

 1. Simeon Vincent (Google)
 2. David Johnson (Apple)
 3. Rob Wu (Mozilla)
 4. Giorgio Maone (NoScript, Tor Project)
 5. Jack Works (Sujitech)
 6. Tim Heflin (Keeper)
 7. Sam Macbeth (DuckDuckGo)
 8. Oliver Dunk (1Password)
 9. Benjamin Bruneau (1Password)
 10. Juha-Matti Santala (Mozilla)
 11. Todd Schiller (PixieBrix)
 12. Tomislav Jovanovic (Mozilla)
 13. Kiara Rose (Apple)
 14. Timothy Hatcher (Apple)
 15. Andrey Meshkov (AdGuard)
 16. Dmitriy Seregin (AdGuard)
 17. Lucas Selker (Dashlane)
 18. Emilia Paz (Google)
 19. Steven McLintock (1Password)
 20. Carlos Jeurissen (Jeurissen Apps)
 21. Tyler Carson (Keeper)
 22. Maxim Topciu (AdGuard)
 23. Mukul Purohit (Microsoft)

## Meeting notes

Introduction

 * [simeon] Today has a different format than usual. Carlos helped with creating the agenda today ([issue 314](https://github.com/w3c/webextensions/issues/314)), instead of our usual Newest issue-first approach.
 * [carlos] It is a long list, it is not a goal to tackle all of them. Prefer to complete topics fully before moving to the next one.

>  * [rob] Aas part of the meeting note publication process, I usually link all discussed topics to the meeting notes PR on Github, which automatically shows up on the Github issues. Given the many topics here, I will delete the ones that have not been discussed to avoid the incorrect impression of an issue having been covered.

Reviewer: Minor typo.

```suggestion
 * [rob] As part of the meeting note publication process, I usually link all discussed topics to the meeting notes PR on Github, which automatically shows up on the Github issues. Given the many topics here, I will delete the ones that have not been discussed to avoid the incorrect impression of an issue having been covered.
```

 * [simeon] In the future we may not have a list of “New issues”.
 * [oliver] We should discuss existing issues that have priority over newly created issues. Some of the new issues are nice-to-have requests.
 * [oliver] Could we stop auto-adding new issues to the agenda?
 * [rob] That risks topics being overlooked. We could auto-add the “needs-agenda” to signal that the topics have not been covered yet.
 * [oliver] We could create a needs-triage label.
 * [rob] Who would triage the labels?
 * [simeon] I will take this task. I have been trying to more consistently comment on new issues.
 * [timothy] I have also been trying to stay on top of issues.

[Issue 244](https://github.com/w3c/webextensions/issues/244#issuecomment-1291174495): New labels: "breaking change", "topic: localization", "topic: action"

>  * [simeon] Topic labels are used as an easy way to group related topics. As we discuss more topics we will probably add more in the future. Any label requests should be discussed in issue 244.

Reviewer: ```suggestion
 * [simeon] Topic labels are used as an easy way to group related topics. As we discuss more topics we will probably add more in the future. Any label requests should be discussed in [issue 244](https://github.com/w3c/webextensions/issues/244).
```

 * [carlos] Edge label is missing?
 * [timothy] Edge label is present.
 * [carlos] But it seems not used.
 * [simeon] We generally ask for explicit support from browser vendors before adding labels. So Mukul, please use the labels where relevant.
 * [mukul] Will do.
 * [rob] Edge is a fork of Chromium, so generally support from Chrome usually implies support in Edge.

[PR 291](https://github.com/w3c/webextensions/pull/291): Document How we work

 * [simeon] We are in a good position to merge these contribution guidelines.
 * [rob] Should I update the PR with what we just agreed on the process first?

>  * [simeon] Let's merge it. Do multiple iterations of smaller updates.

Reviewer: ```suggestion
 * [simeon] Let's merge it now. Want to favor multiple iterations of smaller updates.
```

[Issue 188](https://github.com/w3c/webextensions/issues/188): Inconsistency: Permissions "menus"

 * Check-in on support for replacing contextMenus with menus and (2) changing the API signature of menus.create / contextMenus.create
 * [simeon] I have previously expressed that we do not intend to drop the contextMenus namespace. I will post a comment here.

[Issue 282](https://github.com/w3c/webextensions/issues/282): Proposal: declaring background scripts in a neutral way

 * Check-in on support for declaring bg scripts in a neutral way
 * [simeon] Last night I updated the issue with Chrome's position. We have significant reservations about doing this due to the non-trivial differences between the execution environments. We are open to additional exploration, we are not opposed to a single manifests that works everywhere. We are interested in well-defined semantics. If a developer declares that an extension works in an Event page, a Service worker and a Background page, then it is not obvious whether that is really the case.
 * [rob] The feature request already adds a key to opt in, isn't that a clear enough signal?

>  * [simeon] preferredEnvironment is not a clear enough signal. Maybe it's just a matter of naming.

Reviewer: ```suggestion
 * [simeon] No, we don't think `preferredEnvironment` is not a clear enough signal. That name implies deference rather than an explicit declaration. Maybe it's just a matter of naming.
```

[Issue 160](https://github.com/w3c/webextensions/issues/160): Ensure consistency of action.openPopup API across browsers

 * [oliver] I worked with this with Rob and landed a patch in Firefox. There are some edge cases in issue 160. It would be good to get feedback on.
 * [oliver] One of the issues is what to do when another popup is open. Chrome and Safari behave inconsistently.
 * [timothy] Throwing an error when the panel is already is non-actionable by devs. Suggest no-op.
 * [rob] This is not necessarily about opening the popup itself.
 * [rob] Chrome + Safari: please read the list of inconsistencies in issue 160 and comment on the desired behavior.
 * [oliver] How do we want to turn the discussion from this issue in a specification?

>  * [simeon] I think that makes sense.

Reviewer: ```suggestion
 * [simeon] I think that makes sense to collect the discussion in the issue for future reference when specifying.
```
